python/src/deltachat/chat.py

```diff
@@ -512,7 +512,7 @@ def get_locations(self, contact=None, timestamp_from=None, timestamp_to=None):
                 latitude=lib.dc_array_get_latitude(dc_array, i),
                 longitude=lib.dc_array_get_longitude(dc_array, i),
                 accuracy=lib.dc_array_get_accuracy(dc_array, i),
-                timestamp=datetime.utcfromtimestamp(
+                timestamp=datetime.fromtimestamp(
```

Comment:
I don't really understand the reason for this change here.  Isn't the gossip timestamp stored by core in the UTC timezone?  AFAIK a timestamp is a UNIX epoch which by definition should always represent time as UTC.  I don't think this change the time but it does change the timestamp in the python code local to the timezone instead of i UTC, and that seems and odd choice.

Reply:
You can try
```
python3 -c 'from datetime import datetime; print(datetime.utcfromtimestamp(0).timestamp() / 3600)'
```

It should return 0 for all timezones, but it does not because `.timestamp()` assumes datetime is created with `fromtimestamp`, not `utcfromtimestamp`. This is because there are two different types of datetimes: (timezone-)"aware" and "naive", and "timestamp" assumes the datetime is "aware": https://docs.python.org/3/library/datetime.html#datetime.datetime.timestamp In other words, `timestamp()` method is broken for datetimes returned from `utcfromtimestamp` and documentation encourages switching to `fromtimestamp` because of this.

Reply:
I briefly described it in the second commit message.

Reply:
If we really want to have our datetimes in UTC, we can supply a timezone explicitly. This works correctly:
```
python3 -c 'from datetime import datetime, timezone; print(datetime.fromtimestamp(0, timezone.utc).timestamp() / 3600)'
```
But "naive" datetimes are just broken.

Reply:
oh wow.  python's time handling never stops to amaze, the fact that `.utcfromtimestamp().timestamp()` is not symmetric is... sad.

Though I think the better fix would be to stick with UTC datetime objects but construct them with the timezone info: `datetime.datetime.fromttimestamp(value_from_core, datetime.timezone.utc)`.  With that your roundtrip example does work.

Reply:
Seems like we came to the same conclusion in an async way :)

I'll approve, but would really really prefer you switched to UTC timezone datetime object before merging :)